spirv_glsl.cpp

```diff
@@ -12091,19 +12187,27 @@ void CompilerGLSL::emit_instruction(const Instruction &instruction)
 
 	case OpFRem:
 	{
-		if (is_legacy())
-			SPIRV_CROSS_THROW("OpFRem requires trunc() and is only supported on non-legacy targets. A workaround is "
-			                  "needed for legacy.");
-
 		uint32_t result_type = ops[0];
 		uint32_t result_id = ops[1];
 		uint32_t op0 = ops[2];
 		uint32_t op1 = ops[3];
 
 		// Needs special handling.
 		bool forward = should_forward(op0) && should_forward(op1);
-		auto expr = join(to_enclosed_expression(op0), " - ", to_enclosed_expression(op1), " * ", "trunc(",
-		                 to_enclosed_expression(op0), " / ", to_enclosed_expression(op1), ")");
+		std::string expr;
+		if (!is_legacy())
+			expr = join(to_enclosed_expression(op0), " - ", to_enclosed_expression(op1), " * ", "trunc(",
+			            to_enclosed_expression(op0), " / ", to_enclosed_expression(op1), ")");
+		else
+		{
+			// Legacy GLSL has no trunc, emulate by casting to int and back
```

Comment:
I think it's OK, because the GLSL 1.20 / ESSL 1.00 specs define the behaviour of the int cast as truncating:
> When constructors are used to convert a **float** to an **int**, the fractional part of the floating-point value is dropped.

Alternative is implementing with `floor(abs(x)) * sign(x)` but I think the behaviour of cast is defined well enough as-is (we can revisit if we find actual examples of drivers that disagree).

Thanks for the review, I'll get to work on addressing the other points.